### DISCUSSION TITLE
Syntax for updating unit cell reference frame

### DISCUSSION THREAD

**Discussion by jhiscocks** [mtex-toolbox/mtex#682] [Ask Anything]
Hello All, I realise that you can manually define a crystalSymmetry, but how do you update one without re-inputting it completely?

For example, I can define a magnesium unit cell as follows:
csy2a=crystalSymmetry('6/mmm',[3.2,3.2,5.2],'Y||a','Z||c')

But I want to be able to extract a cs from the ebsd data and then update it.

cs=ebsd('magnesium').CS

Can I update or edit the axes alignment without inputting an entire unit cell?

**Comment by kilir** [comment]
Hi Jessica,
`ebsd.CSList{}` would be the place in your ebsd data. If you have a definition for a crystalSymmetry, you should also be able to change most values in place, however for some assigment ist not supported.


Cheers,
Rüdiger

**Comment by jhiscocks** [comment]
What would be the syntax for changing the reference frame of an existing crystalSymmetry variable?  Or do you need to define a new variable from the ground up?

Thanks, Jessica

**Comment by ralfHielscher** [comment]
Hi Jessica,

currently you indeed need to replace the `crystalSymmetry` variable by a new one, i.e.
```matlab
ebsd('magnesium').CS = crystalSymmetry('622','mineral','magnesium new')
```

We may think of allowing the users to change the setup of the reference frame directly withing the 
`crystalSymmetry` variable. However, this change then would apply globally, i.e., everywhere where this variable has been used the reference system will be changed. This sounds desirable but may also lead to strange side effects.

Ralf.
